src/app/onboarding/onboarding.js

```diff
@@ -94,26 +220,85 @@ angular.module("bawApp.components.onboarding", ["bawApp.citizenScience.common",
+                                onboardingService.callbacks.onBeforeStart[i].call();
+                            }
+                            $timeout(() => {
+                                ngIntroService.start();
+                                onboardingService.started();
+                            });
                         });
 
                     } else {
                         ngIntroService.start();
                     }
+                };
+
+                // automatically start the tour if any of the steps have not been seen yet.
+                // steps are added to the service by different components. It's hard to know when steps have stopped being
+                // added. We check to see if all have been viewed after each addition of steps,
+                // and if false launch the tour after a timeout of 1 second. If new steps are added, this timeout gets extended
+                // if new steps are added after the 1 second, ... hmm I dunno, maybe we need to cancel and relaunch to ensure those steps are included?
+
+                self.autoplayTimeout = null;
+                $scope.$watch(function () {
+                    return onboardingService.isReady();
+                    }, function (isReady) {
+                        if (isReady) {
+                            $scope.status = "ready";
+                            if (!onboardingService.hasViewedAll()) {
+                                ngIntroService.hideHints();
+                                $scope.launchTour();
+                                // $timeout.cancel(self.autoplayTimeout);
+                                // ngIntroService.hideHints();
+                                // self.autoplayTimeout = $timeout(function () {
+                                //     ngIntroService.hideHints();
+                                //     $scope.launchTour();
+                                // }, 500);
+                            }
+                        }
+                });
+
+                /**
+                 * Use the onAfterChange to register that a particular step has been viewed, for the purpose of
+                 * autoplaying only if something has not been viewed.
+                 */
+                onboardingService.addCallbacks({
+                    onAfterChange: function (el) {
+                        onboardingService.registerViewed(ngIntroService.intro._currentStep);
+                    },
+                    onExit: function () {
+                        onboardingService.registerViewed();
+                        self.onClose();
+                    },
+                    onComplete: function () {
+                        self.onClose();
+                    }
+                });
 
+                self.onClose = function () {
+                    $timeout(function () {
+                        $scope.status = "ready";
```

Comment:
I've seen these `status` strings used a few times. Probably should constantize or enumerize them